### DISCUSSION TITLE
Generated pdf previewer in WPF?

### DISCUSSION THREAD

**Discussion by Rayffer** [QuestPDF/QuestPDF#786] [General]
Hello, I have a use case in which, I need to preview the PDF that will be generated inside a WPF application.

The user will use a predefined template by the development team so the user does not need real time edition of the template like the PDF previewer companion offers. 

The user selects content which is provided to the template to generate the PDF, but we usually display a PDF in a window before saving or printing it so that the user can review the data in a final step.

My question is, given my intended use case, which options do I have. The following options are what comes to mind;

1. Start the PDF previewer alongside the application following the usual convention of calling `PreviewPdf` and use an embedded WebBrowser pointing to the PDF previewer endpoint.
2. Generate the PDF anyways with the content and preview it in WPF with a pdf control, which I believe there are some on the market. 
    - In this case I should keep a handle of the file where it is generated and delete it if the user does not proceed forwards or simply directly prints the pdf instead of saving it.

**Comment by user-named-void** [comment]
Hi @Rayffer ,

You could generate images instead of the pdf and use those to preview the document.
You'll get an image for each page.

``` C#
// create the document as you do now, but save it as images
IEnumerable<byte[]> images = Document.Create(container =>
{
    container.Page(page =>
    {
        page.Content()
            .Column(column =>
            {
                // Do the fancy pdf things to generate a nice pdf
            });
    });
}).GenerateImages();  // <-- here you generate the images instead of the pdf.
```

**Comment by Rayffer** [comment]
Will certainly try this in a PoC and report the results back here, thanks for the neat idea

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. We apologize for the delayed response. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
